log4j2-ecs-layout/src/main/java/co/elastic/logging/log4j2/MdcSerializer.java

```diff
@@ -24,65 +24,14 @@
  */
 package co.elastic.logging.log4j2;
 
-import co.elastic.logging.EcsJsonSerializer;
-import co.elastic.logging.JsonUtils;
 import org.apache.logging.log4j.core.LogEvent;
-import org.apache.logging.log4j.util.TriConsumer;
-
-interface MdcSerializer {
 
+/**
+ * Interface for serializing MDC (Mapped Diagnostic Context) data from a {@link LogEvent}.
+ * <p>
+ * Implementations must have a public no-argument constructor to allow dynamic instantiation.
+ * </p>
+ */
+public interface MdcSerializer {
     void serializeMdc(LogEvent event, StringBuilder builder);
```

Comment:
Please add the following javadoc to the method:
```
/**
 * Add MDC data for the give log event to the provided output string builder.
 * The output written to the string builder must be a valid JSON-object without the surrounding curly braces
 * and with a trailing comma. If this MDC serializer does not append any content, no comma shall be added.
 * For example, the serializer could output the following content: "foo":"bar","key":"value",
 * 
 * @param event the log event to write the MDC content for
 * @param builder the output JSON string builder
 */
```

Reply:
Appreciate the comment, I’ve updated it